### PR TITLE
Migrate create post page to mobx + community autocomplete + l10 strings

### LINKED ISSUE #235: Type field for new post commnity
Please add a type instead of a selct field for the community field when you click on the plus button to make a nes post.

The select field is not user friendly as you will have to scroll through hundreds and possibly thousands of options until you find the communuty you want to post in.

### PR DESCRIPTION
**Pull request by shilangyu**
Closes #235

### CODE REVIEW

File: lib/util/files.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import 'dart:io';
+
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter/foundation.dart';
+import 'package:image_picker/image_picker.dart';
+
+/// Picks a single image from the system
+Future<XFile?> pickImage() async {
+  if (kIsWeb || Platform.isIOS || Platform.isAndroid) {
+    return ImagePicker().pickImage(source: ImageSource.gallery);
```
Comment (shilangyu) [inline lib/util/files.dart:10]: Investigate if it works properly on iOS, accessing the path that is returned by ImagePicker is illegal and can randomly fail

File: lib/pages/create_post/create_post_url_field.dart [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+      // pic is null when the picker was cancelled
+      if (pic != null) {
+        await context.read<CreatePostStore>().uploadImage(pic.path, token);
+      }
+    }
+
+    return ObserverConsumer<CreatePostStore>(
+      listener: (context, store) {
+        // needed since flutter's TextFields cannot work as dumb widgets
```
Comment (krawieck) [inline lib/pages/create_post/create_post_url_field.dart:37]: plz edit the comment to make it clear that it's this way so that picture uploading can work

**Review by krawieck — CHANGES_REQUESTED** [review]
most good

File: pubspec.yaml [hunk 3]
```diff
@@ -60,6 +60,8 @@ dependencies:
     sdk: flutter
   flutter_localizations:
     sdk: flutter
+  wc_form_validators: ^1.0.0
+  file_picker: ^4.3.1
```
[the remark this answers is not in the document]
Comment (shilangyu) [reply pubspec.yaml:65]: fixed in https://github.com/LemmurOrg/lemmur/pull/333

File: lib/pages/create_post/create_post_url_field.dart (hunk 2 above)
Comment (shilangyu) [reply lib/pages/create_post/create_post_url_field.dart:37]: This is not true tho, without it picture uploading will work fine. This is only to show on the UI the newest URL

**Review by krawieck — CHANGES_REQUESTED** [review]

**Review by krawieck — APPROVED** [review]

**Pull request merged**